rest/data/generate.go

```diff
@@ -25,28 +27,21 @@ func (gc *GenerateConnector) GenerateTasks(taskID string, jsonBytes []json.RawMe
 	}
 	g := model.MergeGeneratedProjects(projects)
 	g.TaskID = taskID
-	p, v, t, pm, err := g.NewVersion()
-	if err != nil {
-		return err
-	}
-	syntaxErrs, err := validator.CheckProjectSyntax(p)
-	if err != nil {
-		return err
-	}
-	if len(syntaxErrs) > 0 {
-		return gimlet.ErrorResponse{
-			StatusCode: http.StatusBadRequest,
-			Message:    fmt.Sprintf("project syntax is invalid: %s", validator.ValidationErrorsToString(syntaxErrs)),
+
+	for i := 0; i < retryLimit; i++ {
+		p, v, t, pm, prevConfig, err := g.NewVersion()
+		if err != nil {
+			return err
 		}
-	}
-	semanticErrs := validator.CheckProjectSemantics(p)
-	if len(semanticErrs) > 0 {
-		return gimlet.ErrorResponse{
-			StatusCode: http.StatusBadRequest,
-			Message:    fmt.Sprintf("project semantics is invalid: %s", validator.ValidationErrorsToString(semanticErrs)),
+		if err = validator.CheckProjectConfigurationIsValid(p); err != nil {
+			return err
+		}
+		// if config outdated, we try to retry
+		if err = g.Save(p, v, t, pm, prevConfig); err != mgo.ErrNotFound {
+			return err
```

Comment:
The above errors are `gimlet.ErrorResponse`s. This should be too.

Reply:
I think that makes sense for if the error is mgo.ErrNotFound, but otherwise would we return err as we did originally?  I'm not sure what the statusCode would be otherwise.